hypershift-operator/controllers/nodepool/manifests.go

```diff
@@ -45,6 +45,12 @@ func AWSMachineTemplate(infraName, ami string, nodePool *hyperv1.NodePool, contr
 			subnet.Filters = append(subnet.Filters, filter)
 		}
 	}
+	volume := &capiaws.Volume{}
+	if nodePool.Spec.Platform.AWS.RootVolume != nil {
```

Comment:
The implication here is that each of the `RootVolume` fields are immutable, which is something we haven't discussed. Is it the intention that these fields be immutable? If so, we should document that on the API fields.

Can somebody speak to whether these fields could be nondestructively mutable even in theory? I'm not familiar enough with the underlying implementation.

Reply:
I'm sorry, I misread the conditional and see that the fields are being treated as mutable here. Just reverse my prior comments: are these fields actually mutable? If so then I would say there's nothing to document after all.

Reply:
The volume fields aren't something that will be modified frequently. I would imagine someone wants to increase the size, not decrease, and the type and IOPS aren't something that would change either. Is there a reason why we don't want the changes to volume fields to be captured in versioning?

Reply:
What I mean is, are there end-user implications to changing these values? Is it actually possible in AWS to dynamically change the size and type?

For example, what happens if the size is decreased? Is that supported? What if allocation exceeds the new desired size?

Is it actually possible to change the type of a volume once set?

If there are any caveats here, they need described or enforced, otherwise the API becomes surprising (e.g. "I changed this value and nothing happened", or "I changed this value and something broke")

Reply:
According to the [AWS docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-modify-volume.html), it does appear feasible to modify the size, type and performance (IOPS) of an EBS Volume. So I expect they should remain all mutable fields and reconcile/update the changes with CAPI when the user modifies the `NodePool`.

There are these additional notable conditions, imposed by AWS:

> After modifying a volume, you must wait at least six hours and ensure that the volume is in the in-use or available state before you can modify the same volume. This is sometimes referred to as a cooldown period. 

> You can't decrease the size of an EBS volume. 
(From "Limitations" in [the docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/modify-volume-requirements.html)).

Reply:
Thanks, that's useful info!

> According to the [AWS docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-modify-volume.html), it does appear feasible to modify the size, type and performance (IOPS) of an EBS Volume. So I expect they should remain all mutable fields and reconcile/update the changes with CAPI when the user modifies the `NodePool`.
> 
> There are these additional notable conditions, imposed by AWS:
> 
> > After modifying a volume, you must wait at least six hours and ensure that the volume is in the in-use or available state before you can modify the same volume. This is sometimes referred to as a cooldown period.

This one is tricky. Absent any context on the API or explicit behavior in HyperShift to surface status about this sort of thing, how does the end user distinguish between a broken API/bug and a known/expected state? I can mutate the volume and potentially won't observe any changes to the underlying AWS volume right away (for hours!). Will I perceive this as a broken HyperShift?

> > You can't decrease the size of an EBS volume.
> > (From "Limitations" in [the docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/modify-volume-requirements.html)).

Another tricky one. This is an invariant which could be enforced in API validation. Otherwise, as and end user, am I going to perceive the lack of effect of my API change as a bug?

Reply:
My vote is to make these fields immutable (at least in the doc). That's the simplest path forward imo.